packages/stream/create-stream/create-stream.js

```diff
@@ -0,0 +1,114 @@
+ */
+
+/**
+ * @typedef FetchResponse
+ * @property {object} status -- A status response
+ * @property {object} data -- A return data as per endpoint
+ */
+
+/**
+ * A create stream module that should pass stream's name parameter, and user could pass the stream's slug & description parameter as well
+ *
+ * @function
+ * @param {InitializationInstance} initObject -- initialization object
+ * @param {Config} config - passing param from user including name, description, slug
+ * @returns {Promise<FetchResponse>} returns the restructured data which content status & created stream data
+ * @throws {Error}
+ */
+export const createStream = async (initObject, config) => {
+  if (!(initObject instanceof InitializationInstance) || !config) {
+    throw new TypeError(
+      'Failed to process because initialization is not valid and/or missing config. Please provide required initialization argument which is the initialization instance returned by the init() function and the configuration object'
+    )
+  } else if (typeof config !== 'object') {
+    throw new TypeError(
+      'Failed to process because config argument must be in object format'
+    )
+  } else if (config.name === null || config.name === undefined) {
+    throw new Error(
+      'Failed to create a new stream because the name of the stream is empty. Please provide a stream name'
+    )
+  } else if (typeof config.name !== 'string') {
+    throw new TypeError(
+      'Failed to create a new stream because the name of the stream is not in string format. A stream name must be in string format'
+    )
+  } else if (
+    config.slug !== null &&
+    config.slug !== undefined &&
+    typeof config.slug !== 'string'
+  ) {
+    throw new Error(
+      'Failed to create a new stream because the slug of the stream is not in string format. A slug must be in string format'
+    )
+  } else if (
+    config.description !== null &&
+    config.description !== undefined &&
+    typeof config.description !== 'string'
+  ) {
+    throw new Error(
+      'Failed to create a new stream because the description of the stream is not in string format. A description must be in string format'
+    )
+  } else {
+    let fetchResponse = await Internal.fetchHttp({
+      url: `${Internal.config.api.base_url}/${Internal.config.api.version}/streams/create`,
+      token: initObject.config.api_key,
+      method: 'POST',
+      body: {
+        name: config.name,
+        slug: config.slug || replaceWhiteSpace(config.name),
+        description: config.description || '',
+      },
+    }).catch((error) => {
+      return error
+    })
+
+    if (fetchResponse) {
+      switch (fetchResponse.code) {
+        case 200: {
+          fetchResponse = {
+            status: {
+              code: fetchResponse.code,
+              message: 'Successfully created a new stream',
+              type: 'success',
+            },
+            data: fetchResponse?.data,
```

Comment:
The optional chaining `?.` is a new feature from es2020 and has less browser support. We can change this approach using another approach that widely supported by browser

Reply:
@faiq-naufal apologize mas, I forgot to remove `?` . Done remove it at commit [2b610de](https://github.com/inlivedev/inlive-js-sdk/pull/11/commits/2b610de1aaf72e6f2b31ac37f9e68b5b79402d95)